src/main/java/com/leinardi/pycharm/mypy/checker/Problem.java

```diff
@@ -52,17 +52,36 @@ public Problem(final PsiElement target,
         this.suppressErrors = suppressErrors;
     }
 
-    @NotNull
-    public ProblemDescriptor toProblemDescriptor(final InspectionManager inspectionManager) {
-        return inspectionManager.createProblemDescriptor(target,
-                MypyBundle.message("inspection.message", getMessage()),
-                null, problemHighlightType(), false, afterEndOfLine);
+    public void createAnnotation(@NotNull AnnotationHolder holder) {
+        String message = MypyBundle.message("inspection.message", getMessage());
+        AnnotationBuilder annotation = holder
+                .newAnnotation(getHighlightSeverity(), message)
+                .range(target.getTextRange());
+        if (isAfterEndOfLine()) {
+            annotation = annotation.afterEndOfLine();
+        }
+        annotation.create();
     }
 
     public SeverityLevel getSeverityLevel() {
         return severityLevel;
     }
 
+    @NotNull
+    public HighlightSeverity getHighlightSeverity() {
+        switch (severityLevel) {
+            case ERROR:
+                return HighlightSeverity.ERROR;
+            case WARNING:
+                return HighlightSeverity.WARNING;
+            case NOTE:
+                return HighlightSeverity.WEAK_WARNING;
```

Comment:
I think the "weak warning" level is too hard to see

![image](https://user-images.githubusercontent.com/137616/144711931-1b4bbb6a-9219-435d-85a6-e518c05a9db1.png)

I'll revert it to `HighlightSeverity.WARNING`

![image](https://user-images.githubusercontent.com/137616/144712040-87325769-d3c1-483a-af85-6d6412b953a5.png)